packages/contracts-bedrock/src/L1/SuperchainConfig.sol

```diff
@@ -92,4 +119,49 @@ contract SuperchainConfig is Initializable, ISemver {
         Storage.setAddress(GUARDIAN_SLOT, _guardian);
         emit ConfigUpdate(UpdateType.GUARDIAN, abi.encode(_guardian));
     }
+
+    /// @notice Adds a new chain to the dependency set.
+    ///         Adds the new chain as a dependency for all existing chains in the dependency set, and vice versa. It
+    ///         also stores the SystemConfig address of it, and authorizes the OptimismPortal on the SharedLockbox.
+    /// @param _chainId     The chain ID.
+    /// @param _systemConfig The address of the SystemConfig contract.
+    function addChain(uint256 _chainId, address _systemConfig) external {
+        // TODO: Updater role TBD, using guardian for now.
+        if (msg.sender != guardian()) revert Unauthorized();
+        // Add to the dependency set and check it is not already added (`add()` returns false if it already exists)
+        if (!_dependencySet.add(_chainId)) revert ChainAlreadyAdded();
+
+        systemConfigs[_chainId] = _systemConfig;
+
+        // If the dependency set is empty, there is no need to update the dependencies
+        if (_dependencySet.length() != 0) {
+            // Loop through the dependency set and update the dependency for each chain. Using length - 2 to exclude the
+            // current chain from the loop.
+            for (uint256 i; i < _dependencySet.length() - 1; i++) {
+                uint256 currentId = _dependencySet.at(i);
+
+                // Add the new chain as dependency for the current chain on the loop
+                ISystemConfigInterop(systemConfigs[currentId]).addDependency(_chainId);
+                // Add the current chain on the loop as dependency for the new chain
+                ISystemConfigInterop(_systemConfig).addDependency(currentId);
+            }
+        }
+
+        // Authorize the portal on the shared lockbox
+        address portal = ISystemConfigInterop(_systemConfig).optimismPortal();
+        ISharedLockbox(sharedLockbox).authorizePortal(portal);
+
+        emit ChainAdded(_chainId, _systemConfig, portal);
+    }
+
+    /// @notice Checks if a chain is part or not of the dependency set.
+    /// @param _chainId The chain ID to check for.
+    function isInDependencySet(uint256 _chainId) public view returns (bool) {
```

Comment:
should it extend from `IDependencySet` for this function or it doesnt add any value?